### ISSUE TITLE
/_matrix/consent fails with 500 error

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#3731]
```
2018-08-21 21:29:30,810 - synapse.http.server - 162 - ERROR - GET-855803 - Failed handle request <XForwardedForRequest at 0x7fb85fdeb518 method='GET' uri='/_matrix/co
  File "synapse/http/server.py", line 126, in wrapped_request_handler
    
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
    result = f(*args, **kw)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
    return _cancellableInlineCallbacks(gen)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
    _inlineCallbacks(None, g, status)
--- <exception caught here> ---
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "synapse/rest/consent/consent_resource.py", line 145, in _async_render_GET
    
  File "synapse/rest/consent/consent_resource.py", line 219, in _check_hash
    
exceptions.TypeError: 'unicode' does not have the buffer interface
```

**Comment by richvdh** [comment]
This was introduced in #3668

**Comment by richvdh** [comment]
and fixed in #3732
